processing/data_collection/gazette/pipelines.py

```diff
@@ -64,44 +43,64 @@ def process_item(self, item, spider):
         return item
 
 
-class DocToPdfPipeline:
+class ExtractTextPipeline:
     """
-    Convert a doc[x] file to pdf
+    Identify file format and call the right tool to extract the text from it
     """
 
     def process_item(self, item, spider):
-        # if item is not a doc, skip it
-        if not self.is_doc(item["files"][0]["path"]):
-            return item
-        # it's doc[x]. Convert it to pdf
-        doc_path = os.path.join(FILES_STORE, item["files"][0]["path"])
-        # use libreoffice writer to convert
-        command = f"lowriter --convert-to pdf --outdir {FILES_STORE}/full {doc_path}"
-        subprocess.run(command, shell=True, check=True)
-        pdf_path = doc_path + ".pdf"
-        # update to the new file path and its checksum
-        item["files"][0]["path"] = pdf_path
-        item["files"][0]["checksum"] = self.calculate_md5sum(pdf_path)
+        if self.is_doc(item["files"][0]["path"]):
+            item["source_text"] = self.doc_source_text(item)
+        elif self.is_pdf(item["files"][0]["path"]):
+            item["source_text"] = self.pdf_source_text(item)
+        else:
+            raise "Unsupported file type"
+
+        for key, value in item["files"][0].items():
+            item[f"file_{key}"] = value
+        item.pop("files")
+        item.pop("file_urls")
         return item
 
+    def pdf_source_text(self, item):
+        """
+        Gets the text from pdf files
+        """
+        pdf_path = os.path.join(FILES_STORE, item["files"][0]["path"])
+        command = f"pdftotext -layout {pdf_path}"
+        subprocess.run(command, shell=True, check=True)
+        if ".pdf" in pdf_path:
+            text_path = pdf_path.replace(".pdf", ".txt")
+        else:
+            text_path = pdf_path + ".txt"
```

Comment:
Similarly to how you did in `doc_source_text()`, I think here would be better to just append `.txt` to the filename, getting something like `foobar.pdf.txt`, as @cuducos suggested this would keep the information that this file was originally a PDF.

Reply:
Done! ;)